tests/p2p/test_sync.py

```diff
@@ -305,6 +310,142 @@ def test_downloader(self):
         downloader.check_downloading_queue()
         self.assertEqual(len(downloader.downloading_deque), 0)
 
+    def test_downloader_retry_reorder(self):
+        """ Reproduce the bug that causes a reorder in the downloader queue.
+
+        The tracking issue for this bug is #465
+
+
+        In order for the bug to be triggered, the following events must happen in the following order:
+
+        - premise:
+          - be connected to two nodes which are ahead of our node (we'll call them peer_X and peer_Y)
+          - sync timestamp must have requested two transactions, tx_A and tx_B, of which tx_B depends on tx_A (tx_A can
+            be parent of tx_B)
+        - while tx_A and tx_B are in the downloader, these are the key events that trigger the issue:
+          - tx_A is requested for download to peer_X
+          - tx_B is requested for download to peer_X
+          - peer_X disconnects
+          - peer_Y disconnects
+          - download of tx_A timeouts, since there are no nodes to download it from it is removed from the downloader,
+            and it is not retried
+          - peer_X re-connects, sync starts and tx_A and tx_B are added to the downloader
+          - peer_Y re-connects, sync starts and tx_A and tx_B are added to the downloader
+          - download of tx_B timeouts, since there now are peers to download it from, it isn't removed and a retry is
```

Comment:
Typo: "since there are peers available to download it from".

Reply:
Fixed.